### DISCUSSION TITLE
Multiple prefixes in one app module for index db

### DISCUSSION THREAD

**Discussion by EnoRage** [cyrilletuzi/angular-async-local-storage#559] [General]
## Description 
I want to have several services in the same module with different namespace for indexedDB

## Reason
 I have different services responsible for storing charts data and some other operations data. I have already store data at default prefix ('local-storage')

## What I expect
I was trying to create an own instance of storage like:

```TS
this.storage = new StorageMap(new IndexedDBDatabase(), new JSONValidator(), 'charts');
```

## What I got
IndexedDBDatabase is not exportable for now. Is there any way to do it without touching app module.

## What I want
To create another service with different prefix for indexDB in the same module

**Comment by cyrilletuzi** [comment]
Hi,

This feature doesn't not exist, and I currently have no availability to work on that. Current possibilities are:
- a community PR, *after having agreed on the API design*
- [sponsoring](https://github.com/sponsors/cyrilletuzi/)
- use the advanced API to simulate the feature ([recipe here](https://github.com/cyrilletuzi/angular-async-local-storage/blob/main/docs/MAP_OPERATIONS.md))
